src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx

```diff
@@ -225,13 +225,8 @@ export const MyCollectionArtworkForm: React.FC<MyCollectionArtworkFormProps> = (
     }
   }
 
-  const leaveForm = () => {
-    router.replace({ pathname: "/settings/my-collection" })
-    router.push({
-      pathname: isEditing
-        ? `/my-collection/artwork/${artwork.internalID}`
-        : "/settings/my-collection",
-    })
+  const goBack = () => {
+    router.go(-1)
```

Comment:
@dzucconi @damassi This is a terrible hack but it does the trick for me, is there a way to do it better?
(I didn't change the breaking tests for it yet)

<details>
<summary>Context</summary>

This page can be accessed from multiple places (Adding new artwork to My Collection), and we would like to have the back component to navigate back to where the user came from, either My Collection list page or Insights page
</details>

Reply:
I think that's fine / not a hack, really. What happens if someone hits this URL directly without having a referrer?

Reply:
The browser sends the user back to the previous page, whether it was on Artsy or not